gr-blocks/lib/interleaved_char_to_complex_impl.cc

```diff
@@ -38,10 +43,12 @@ int interleaved_char_to_complex_impl::work(int noutput_items,
                                            gr_vector_const_void_star& input_items,
                                            gr_vector_void_star& output_items)
 {
-    const int8_t* in = (const int8_t*)input_items[0];
     float* out = (float*)output_items[0];
+    const int8_t* in = (const int8_t*)input_items[0];
 
-    volk_8i_s32f_convert_32f_u(out, in, 1.0, 2 * noutput_items);
+    // This calculates in[] * 1.0 / d_scalar
+    volk_8i_s32f_convert_32f(
+        out, in, d_scalar, d_vector ? 4 * noutput_items : 2 * noutput_items);
```

Comment:
should be `2 * noutput_items` either way again, yes?

Reply:
also: do we want to add in a `swap` like the `short` version has? I honestly don't know why this block doesn't have the option.

Reply:
Here's one way to look at this question: suppose we want 1 output item. If vector input, this requires 1 input item (a vector of length 2), for a total of 2 `char` inputs; if not vector, this requires 2 input items, for a total of 2 `char` inputs. Either way, we require a total of 2 `char` inputs to generate this 1 output item. So the number of items for `volk_8i_s32f_convert_32f` to process is 2 for this specific situation. Hence we always require `2 * noutput_items` input `char` to generate `noutput_items` outputs. Yes?

Reply:
_thou must count to 2. 2 shall be the number of the counting and the number of the counting shall be 2. 3 shalt thou not count, neither shalt thou count 1, excepting that thou then proceedeth to 2. 4 is right out._